library/std/src/net/ip.rs

```diff
@@ -1488,6 +1492,33 @@ impl Ipv6Addr {
         (self.segments()[0] & 0xff00) == 0xff00
     }
 
+    /// Converts this address to an [IPv4 address]. Returns [`None`] if this address is
+    /// not IPv4-mapped.
+    ///
+    /// ::ffff:a.b.c.d become a.b.c.d
+    ///
+    /// [IPv4 address]: ../../std/net/struct.Ipv4Addr.html
+    /// [`None`]: ../../std/option/enum.Option.html#variant.None
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::net::{Ipv4Addr, Ipv6Addr};
+    ///
+    /// assert_eq!(Ipv6Addr::new(0xff00, 0, 0, 0, 0, 0, 0, 0).to_ipv4_mapped(), None);
+    /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).to_ipv4_mapped(),
+    ///            Some(Ipv4Addr::new(192, 10, 2, 255)));
+    /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1).to_ipv4_mapped(), None);
+    /// ```
+    pub fn to_ipv4_mapped(&self) -> Option<Ipv4Addr> {
```

Comment:
```suggestion
    #[unstable(feature = "ipv4_mapped", issue = none)]
    pub fn to_ipv4_mapped(&self) -> Option<Ipv4Addr> {
```

Reply:
thx~

Reply:
Do I need to add `#![feature(ipv4_mapped)]` to the lib.rs?

Reply:
Yes, certainly.

Reply:
It's done.

Reply:
You still need to add the unstable attribute.

Reply:
Although there is already an unstable feature named `ip`, do I still need to add a new unstable feature?

#27709 
https://github.com/rust-lang/rust/blob/master/library/std/src/net/ip.rs#L1-L7

Reply:
I [asked on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/Adding.20new.20methods.20to.20existing.20features.3F) and it looks like adding stuff to existing unstable features is fine. So just use the `ip` feature :)

Reply:
Since you requested another review: the `unstable` attribute is still missing. Please add `#[unstable(feature = "ip", issue = "27709")]` right above the line containing `pub fn`.

Reply:
Since the following lines have been added at the beginning of this file (`ip.rs`):
```rust
#![unstable(
    feature = "ip",
    reason = "extra functionality has not been \
                                      scrutinized to the level that it should \
                                      be to be stable",
    issue = "27709"
)]
```
Adding `#[unstable(feature = "ip", issue = "27709")]` above this fn is unnecessary.

Reply:
(It is better to give a permanent link to those code lines in the future).